### PR TITLE
woking on the user and admin

### PR DESCRIPTION
**Pull request by chibukasia**

### CODE REVIEW

File: app/controllers/admins_controller.rb [hunk 1]
```diff
@@ -0,0 +1,32 @@
+class AdminsController < ApplicationController
+  def index
+    admin = Admin.all
```
Comment (chibukasia) [inline app/controllers/admins_controller.rb]: Pluralize admin to admins

File: app/controllers/admins_controller.rb [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+    render json: admin, status: :ok
+  end
+
+  def create
+    admin = Admin.create!(admin_params)
```
Comment (chibukasia) [inline app/controllers/admins_controller.rb]: Use create without exclamation for now.

File: app/controllers/admins_controller.rb [hunk 3]
```diff
@@ -0,0 +1,32 @@
...
+    render json: admin, status: :created
+  end
+
+  def update
+    admin = Admin.update(admin_params)
```
Comment (chibukasia) [inline app/controllers/admins_controller.rb]: Use a find private method to find the user to update. 
Use update without exclamation for now

File: app/controllers/admins_controller.rb [hunk 4]
```diff
@@ -0,0 +1,32 @@
...
+    render json: admin, status: :ok
+  end
+
+  def show
```
Comment (chibukasia) [inline app/controllers/admins_controller.rb:25]: Create a private method for finding a admin 
Create a private method for params permit

File: app/controllers/admins_controller.rb (hunk 1 above)
Comment (munenebrian) [reply app/controllers/admins_controller.rb]: done

File: app/controllers/admins_controller.rb (hunk 2 above)
Comment (munenebrian) [reply app/controllers/admins_controller.rb]: checked and done

File: app/controllers/admins_controller.rb (hunk 3 above)
Comment (munenebrian) [reply app/controllers/admins_controller.rb]: reviewed and implemented changes

File: app/controllers/admins_controller.rb (hunk 4 above)
Comment (munenebrian) [reply app/controllers/admins_controller.rb:25]: done

**Comment by munenebrian** [comment]
I have checked on the comments and have made the pushes

File: app/controllers/users_controller.rb [hunk 5]
```diff
@@ -1,32 +1,55 @@
...
+  #the private methods
+  private
+
+  #find user
+  def find_user
+    User.find(params(:id))
+  end
+
+  #allowed params
+  def user_params
```
Comment (chibukasia) [inline app/controllers/users_controller.rb:46]: When creating a user, we take the params, first_name, last_name, username, email, phone_number, role, password, password_confirmation. The parameters permitted are misinterpreted and others are missing
We need to have something like `params.permit(:first_name, :last_name, :username, :email, :phone_number, :role, :password, :password_confirmation)`

File: app/controllers/admins_controller.rb [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+  private
+
+  #find the admin
+  def find_admin
+    Admin.find(params(:id))
+  end
+
+  allowed params
+  def admin_params
+    params.permit(:admin_name)
```
Comment (chibukasia) [inline app/controllers/admins_controller.rb]: Same to users, you missed adding all the admin params. The admins params are username, password, password_confirmation. It should be this way `params.permit(:username, :password, :password_confirmation)`
Kndly make these few changes

**Pull request merged**